vllm/model_executor/layers/fused_moe/moe_permute_unpermute.py

```diff
@@ -0,0 +1,119 @@
+import torch
+from vllm import _custom_ops as ops
+from typing import Optional
+
+
+def moe_permute(
+    hidden_states: torch.Tensor,
+    topk_weights: torch.Tensor,
+    topk_ids: torch.Tensor,
+    token_expert_indices: torch.Tensor,
+    topk: int,
+    n_expert: int,
+    n_local_expert:int,
+    expert_map: Optional[torch.Tensor] = None,
+    align_block_size:Optional[int]=None,
+    fill_invalid_expert:int = -1
+) -> list[torch.Tensor]:
```

Comment:
Is there any reason not to make the return type an explicit tuple of tensors?  So users will know exactly how many outputs to expect?